libs/designer/src/lib/ui/mcp/operations/ListOperations.tsx

```diff
@@ -23,13 +59,87 @@ export const ListOperations = ({ connectorId }: { connectorId: string }) => {
     [dispatch]
   );
 
+  const handleDeleteOperation = useCallback(
+    (operationId: string) => {
+      dispatch(deinitializeOperationInfo({ id: operationId }));
+      dispatch(deinitializeNodes([operationId]));
+    },
+    [dispatch]
+  );
+
+  const items = useMemo(() => {
+    return Object.values(operationInfos)
+      .filter((info) => Boolean(info?.operationId))
+      .map((info) => ({
+        operationId: info.operationId,
+        operationName: operationMetadata[info.operationId]?.summary ?? info.operationId,
+        connectorId: info.connectorId,
+      }));
+  }, [operationInfos, operationMetadata]);
+
+  const columns = [
+    { columnKey: 'operation', label: INTL_TEXT.operationLabel },
+    { columnKey: 'actions', label: '' }, // Empty label for actions column
+  ];
+
+  if (!items.length) {
+    return null;
+  }
+
+  if (isInitializingOperations) {
+    return (
+      <div className={styles.emptyState}>
+        <Spinner size="medium" label={INTL_TEXT.loadingOperationsText} />
+      </div>
+    );
+  }
+
   return (
-    <div>
-      {operationInfosWithConnectorId?.map((operationInfo) => (
-        <Button key={operationInfo?.operationId} onClick={() => handleOperationDataClick(operationInfo?.operationId)}>
-          {operationInfo?.operationId}
-        </Button>
-      ))}
-    </div>
+    <Table className={styles.tableStyle} aria-label={INTL_TEXT.tableAriaLabel} size="small">
+      <TableHeader>
+        <TableRow style={{ border: 'none' }}>
+          {columns.map((column) => (
+            <TableHeaderCell key={column.columnKey}>
+              <Text weight="semibold">{column.label}</Text>
+            </TableHeaderCell>
+          ))}
+        </TableRow>
+      </TableHeader>
+      <TableBody>
+        {items.map((item) => (
+          <TableRow key={item.operationId}>
+            <TableCell className={styles.iconTextCell}>
+              <ConnectorIconWithName
+                classes={{
+                  root: 'msla-template-create-connector',
+                  icon: 'msla-template-create-connector-icon',
+                  text: 'msla-template-create-connector-text',
+                }}
+                connectorId={item.connectorId}
+                operationId={item.operationId}
+                onNameClick={() => handleEditOperation(item.operationId)}
+              />
```

Comment:
Why are we using this to list the operation? I dont think you should be passing the operationId here, it is going to try get the operation manifest because you are providing a tuple... do not set the operationId...
Also Why are we sending the operationId here, shouldnt we be getting the summary from store and just be rending hte icon we got from store and render it... there is no need to do get connector call here no?